gr-utils/python/blocktool/core/parseheader.py

```diff
@@ -0,0 +1,281 @@
+                                                    header_file=self.target_file)
+            criteria = declarations.calldef_matcher(name='make')
+            _make_fun = declarations.matcher.get_single(criteria, main_class)
+            _make_fun = str(_make_fun).split(
+                'make')[-1].split(')')[0].split('(')[1].lstrip().rstrip().split(',')
+            if make_func:
+                for arg in make_func[0].arguments:
+                    for _arg in _make_fun:
+                        if str(arg.name) in _arg:
+                            make_arguments = {
+                                "name": str(arg.name),
+                                "dtype": str(arg.decl_type),
+                                "default": ""
+                            }
+                            if re.findall(r'[-+]?\d*\.\d+|\d+', _arg):
+                                make_arguments['default'] = re.findall(
+                                    r'[-+]?\d*\.\d+|\d+', _arg)[0]
+                            elif re.findall(r'\"(.+?)\"', _arg):
+                                make_arguments['default'] = re.findall(
+                                    r'\"(.+?)\"', _arg)[0]
+                            elif "true" in _arg:
+                                make_arguments['default'] = "True"
+                            elif "false" in _arg:
+                                make_arguments['default'] = "False"
+                    self.parsed_data['make']['arguments'].append(
+                        make_arguments.copy())
+        except RuntimeError:
+            self.parsed_data['make'] = {}
+            self.parsed_data['make']['arguments'] = []
+
+        # setters
+        try:
+            self.parsed_data['methods'] = []
+            query_methods = declarations.access_type_matcher_t('public')
+            setters = main_class.member_functions(function=query_methods,
+                                                  allow_empty=True,
+                                                  header_file=self.target_file)
+            getter_arguments = []
+            if setters:
+                for setter in setters:
+                    if str(setter.name).startswith('set_') and setter.arguments:
+                        setter_args = {
+                            "name": str(setter.name),
+                            "arguments_type": []
+                        }
+                        for argument in setter.arguments:
+                            args = {
+                                "name": str(argument.name),
+                                "dtype": str(argument.decl_type)
+                            }
+                            getter_arguments.append(args['name'])
+                            setter_args['arguments_type'].append(args.copy())
+                        self.parsed_data['methods'].append(setter_args.copy())
+            else:
+                self.parsed_data['methods'] = []
+        except RuntimeError:
```

Comment:
so the result from not having `setters` (or `getters` for lines 238 on) is the same as when there is a `RuntimeError`. Is this on purpose? Any specific reason not to re-raise with your custom exception and/or notify that an exception happened?

Reply:
Yeah. The else statement is not required. I have made the changes.